src/server/RedirectingHttpHandler.ts

```diff
@@ -0,0 +1,65 @@
+import { getLoggerFor } from '../logging/LogUtil';
+import { NotImplementedHttpError } from '../util/errors/NotImplementedHttpError';
+import type { HttpHandlerInput } from './HttpHandler';
+import { HttpHandler } from './HttpHandler';
+
+export class RedirectingHttpHandler extends HttpHandler {
+  private readonly logger = getLoggerFor(this);
+  private readonly matcher: RegExp;
+  private readonly redirects: { pattern: RegExp; redirect: string }[] = [];
+
+  /**
+   * Creates a handler for the provided redirects.
+   * @param redirects - A mapping between URL patterns.
+   */
+  public constructor(redirects: Record<string, string>, private readonly statusCode: number = 308) {
```

Comment:
I just realized we already have a class that does something similar: the `RedirectAllHttpHandler`. That one specifically redirects all request to one specific URL, so it is a more specific implementation of this class. Looking at that class I see that we do use a `ResponseWriter` to write the response, which makes sense so we handle all responses in a single class. So I want to partially retract my answer below about how to write responses and would prefer it to happen like in the `RedirectAllHttpHandler`. It also uses the `baseUrl` to generate the URLs which is actually necessary as the implementation suggested here would not work with non-root base URLs.

Besides that I think this class should replace the `RedirectAllHttpHandler` in this PR so we don't end up with multiple similar classes (this would require it to target the 4.0.0 branch due to the removal of that class). The one instance where `RedirectAllHttpHandler` is used could then be implemented by having a redirect key `.*` and value `/setup`.

Reply:
> Any idea what causes this weird behavior?

So for some reason when doing this Components.js converts the array `[ "/setup" ]` to `"/setup` here: https://github.com/CommunitySolidServer/CommunitySolidServer/blob/5e6c42e1f82f3474c64c8f43ef1827775ce8cfa7/config/app/setup/required.json#L24

But only when that string matches the one in the `RedirectingHttpHandler`. I have seen some weird issues with arrays in Components.js but this one is the most special case so far 😄 .  Components.js v5 fixes all of the array issues so potentially it would not be an issue there, but since we're still on v4 for now I'll see if I somehow can find a workaround.

Reply:
At this point I'm thinking the easiest solution might be to remove the `RedirectAllHttpHandler` and add a note to the Components.js v5 issue to do it after that so hold off on changing to the 4.0.0 branch should you be looking into that.

Reply:
So you're saying: add the new handler, but also leave the `RedirectAllHttpHandler` for now, so we can merge into `main`, then do the switch when Components.js is bumped?

Reply:
I'll play around with it for a bit to see if it's maybe another cause I'm missing but that's what I'm suggesting yes.

Reply:
There are some workaround possible but they require modifying other code which will make things uglier so would just proceed as above. I'll look into this then once we upgrade components.js.

Reply:
Okay, I've reverted the commit that did the replacement.